lls/main.c

```diff
@@ -821,6 +890,50 @@ register_nd_neigh_acl_rules(struct gatekeeper_if *iface)
 	return 0;
 }
 
+static void
+fill_ping_rule(struct ipv4_acl_rule *rule, struct in_addr *addr)
+{
+	rule->data.category_mask = 0x1;
+	rule->data.priority = 1;
+	/* Userdata is filled in in register_ipv4_acl(). */
+
+	rule->field[PROTO_FIELD_IPV4].value.u8 = IPPROTO_ICMP;
+	rule->field[PROTO_FIELD_IPV4].mask_range.u8 = 0xFF;
+
+	rule->field[DST_FIELD_IPV4].value.u32 =
+		rte_be_to_cpu_32(addr->s_addr);
+	rule->field[DST_FIELD_IPV4].mask_range.u32 = 32;
+
+	/*
+	 * The reason to have 0xFFFF0000 is to test
+	 * both type and code at the same time.
+	 */
+	rule->field[TYPE_FIELD_ICMP].value.u32 =
+		((ICMP_ECHO_REQUEST_TYPE << 24) |
+		(ICMP_ECHO_REQUEST_CODE << 16)) & 0xFFFF0000;
+	rule->field[TYPE_FIELD_ICMP].mask_range.u32 = 0xFFFF0000;
+}
+
+static int
+register_ping_acl_rules(struct gatekeeper_if *iface)
+{
+	struct ipv4_acl_rule ipv4_rule;
```

Comment:
Let's define `ipv4_rule` as an array of size 1 to make this function analogous to its sister functions and easier to eventually edit it later. Rename `ipv4_rule` to `ipv4_rules`.